keps/sig-apps/3715-indexed-job-with-unset-completions/README.md

```diff
@@ -0,0 +1,896 @@
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [x] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Currently Jobs are validated to have `spec.completions` set when `spec.completionMode`=`Indexed`. This KEP proposes to relax this validation and allow 
+Indexed jobs to be created with zero `spec.completions`. 
+
+When `spec.completions` is not set for an `Indexed` Job, the behavior is summarized as follows:
+- The success, failure and finish semantics are same as `NonIndexed`. Specifically:
+  - Success semantics:
+    - The job succeeded if any pod exits successfully. 
+    - The remaining active pods are allowed to continue to run to completion, and the job is marked as finished only 
+      when all pods complete. The success status doesn't change if any of the remaining pods fail, and they will not be retried.
```

Comment:
The comments in the code say that failures are ignored, this may not actually be true, it seems backoffLimit is tied to finish status; if so, then pod failures will count against backoffLimit, I am still trying to verify by reading the code.

Reply:
Updated the KEP, once a pod exists successfully, no new pods will be created, but failures count against the backoffLimit.